### PR TITLE
New artifacts

### PR DESCRIPTION
**Pull request by agr**
Made `NuGet.Jobs.Common` and `Validation.Common.Job` to produce .nupkg's so they could be reused in projects in other repos.

Progress on https://github.com/NuGet/Engineering/issues/1280

### CODE REVIEW

File: src/Validation.Common.Job/Validation.Common.Job.nuspec [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<package >
+  <metadata>
+    <id>NuGet.Services.Validation.Common.Job.$branch$</id>
```
Comment (loic-sharma) [inline src/Validation.Common.Job/Validation.Common.Job.nuspec]: I wouldn't put the $branch$ in the id as this is really annoying to work with on deployments (you have to select the proper nupkg to deploy).

File: src/Validation.Common.Job/Validation.Common.Job.nuspec [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    <version>$version$</version>
+    <title>NuGet.Services.Validation.Common.Job</title>
+    <authors>.NET Foundation</authors>
+    <owners>.NET Foundation</owners>
+    <description>NuGet.Services.Validation.Common.Job</description>
+    <copyright>Copyright .NET Foundation</copyright>
+    <dependencies>
+        <dependency id="NuGet.Jobs.Common" version="$version$" />
```
Comment (joelverhagen) [inline src/Validation.Common.Job/Validation.Common.Job.nuspec]: Spoke offline. I think we're missing some dependencies here. The produced .nupkg... does it have what it needs?

File: src/Validation.Common.Job/Validation.Common.Job.nuspec (hunk 1 above)
Comment (agr) [reply src/Validation.Common.Job/Validation.Common.Job.nuspec]: Done

File: src/Validation.Common.Job/Validation.Common.Job.nuspec [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        <dependency id="Autofac" version="4.6.2" />
+        <dependency id="Autofac.Extensions.DependencyInjection" version="4.2.0" />
+        <dependency id="Microsoft.ApplicationInsights" version="2.2.0" />
+        <dependency id="Microsoft.Extensions.DependencyInjection" version="1.1.1" />
+        <dependency id="Microsoft.Extensions.Options.ConfigurationExtensions" version="1.1.2" />
+        <dependency id="NuGet.Packaging" version="4.7.0-preview1-4886" />
+        <dependency id="NuGet.Services.Configuration" version="2.19.1" />
```
Comment (joelverhagen) [inline src/Validation.Common.Job/Validation.Common.Job.nuspec]: Can we have some kind of automation (maybe a hacky unit test) that compares this with the project dependencies?

Comment (joelverhagen) [reply src/Validation.Common.Job/Validation.Common.Job.nuspec]: I think we need _some_ kind of mitigation to help prevent these things from diverging...

Comment (agr) [reply src/Validation.Common.Job/Validation.Common.Job.nuspec]: Done

File: tests/Validation.Common.Job.Tests/Validation.Common.Job.Tests.csproj [hunk 4]
```diff
@@ -64,5 +65,8 @@
       <Name>Validation.PackageSigning.Core.Tests</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
```
Comment (joelverhagen) [inline tests/Validation.Common.Job.Tests/Validation.Common.Job.Tests.csproj]: Necessary?

File: tests/Validation.Common.Job.Tests/CsprojNuspecConsistencyFacts.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using Xunit;
+
+namespace Validation.Common.Job.Tests
+{
+    public class CsprojNuspecConsistencyFacts
+    {
+        [Fact]
+        public void CsprojReferencesMatchNuspec()
```
Comment (joelverhagen) [inline tests/Validation.Common.Job.Tests/CsprojNuspecConsistencyFacts.cs]: 🥇

File: tests/Validation.Common.Job.Tests/Validation.Common.Job.Tests.csproj (hunk 4 above)
Comment (agr) [reply tests/Validation.Common.Job.Tests/Validation.Common.Job.Tests.csproj]: No idea. Was added automagically. Will remove and see if added again.

Comment (agr) [reply tests/Validation.Common.Job.Tests/Validation.Common.Job.Tests.csproj]: https://stackoverflow.com/a/22917902/31782 ?

Comment (agr) [reply tests/Validation.Common.Job.Tests/Validation.Common.Job.Tests.csproj]: I guess it will be added back eventually.

**Pull request merged**